packages/discord.js-utilities/src/lib/PaginatedMessages/PaginatedFieldMessageEmbed.ts

```diff
@@ -0,0 +1,166 @@
+	 * ```
+	 */
+	public setTemplate(template: MessageEmbedOptions | MessageEmbed) {
+		this.embedTemplate = new MessageEmbed(template);
+		return this;
+	}
+
+	/**
+	 * The way in which the items of the object array will be represented in the embed field to be paginated.
+	 *
+	 * @example
+	 * ```typescript
+	 * import { PaginationFieldMessageEmbed } from '@sapphire/discord.js-utilities';
+	 *
+	 * new PaginationFieldMessageEmbed()
+	 *    .setTitleField('Test field')
+	 *    .setTemplate({ embed })
+	 *    .setItems([
+	 *       { title: 'Sapphire Framework', value: 'discord.js Framework' },
+	 *       { title: 'Sapphire Framework 2', value: 'discord.js Framework 2' },
+	 *       { title: 'Sapphire Framework 3', value: 'Discord.js Framework 3' }
+	 *     ])
+	 *    .formatField((item) => `${item.title}\n${item.value}`)
+	 *    .make()
+	 *    .run(message);
+	 * ```
+	 * @param value Item information
+	 */
+	public formatField(value: (value: T, index: number, array: T[]) => any) {
+		this.items = this.items.map(value);
+		return this;
+	}
+
+	/**
+	 * Build the item pagination of the configured array.
+	 * It is required to put the `.make()` and `.run()` method at the end in order to start.
+	 *
+	 * @example
+	 * ```typescript
+	 * import { PaginationFieldMessageEmbed } from '@sapphire/discord.js-utilities';
+	 *
+	 * new PaginationFieldMessageEmbed()
+	 *    .setTitleField('Test field')
+	 *    .setTemplate({ embed })
+	 *    .setItems([
+	 *       { title: 'Sapphire Framework', value: 'discord.js Framework' },
+	 *       { title: 'Sapphire Framework 2', value: 'discord.js Framework 2' },
+	 *       { title: 'Sapphire Framework 3', value: 'Discord.js Framework 3' }
+	 *     ])
+	 *    .formatField((item) => `${item.title}\n${item.value}`)
+	 *    .make()
+	 *    .run(message);
+	 * ```
+	 */
+	public make() {
+		if (!this.fieldTitle.length) throw new Error('The title of the field to format must have a value.');
+		if (!this.items.length) throw new Error('The array introduced has no items.');
+		if (!this.items[0]) throw new Error('The format of the array items is incorrect.');
+
+		this.totalPages = Math.ceil(this.items.length / this.itemsPerPage);
+		this.makePages();
+		return this;
+	}
+
+	private makePages() {
+		new Array(this.totalPages).forEach((_, index) => {
+			const clonedTemplate = this.embedTemplate instanceof MessageEmbed ? this.embedTemplate : new MessageEmbed(this.embedTemplate);
+			const fieldsClone = this.embedTemplate.fields;
+			clonedTemplate.fields = [];
+
+			if (!clonedTemplate.footer) clonedTemplate.setFooter(`${index + 1}/${this.totalPages}`);
+			if (!clonedTemplate.color) clonedTemplate.setColor('RANDOM');
+
+			const data = this.paginateArray(this.items, index + 1, this.itemsPerPage);
+			this.addPage({
+				embeds: [clonedTemplate.addField(this.fieldTitle, data.join('\n'), false).addFields(fieldsClone)]
+			});
+		});
```

Comment:
```suggestion
		for (let i = 0; i < this.totalPages; i++) {
			const fieldsClone = this.embedTemplate.fields;
			this.embedTemplate.fields = [];

			if (!this.embedTemplate.footer) this.embedTemplate.setFooter(`${i + 1}/${this.totalPages}`);
			if (!this.embedTemplate.color) this.embedTemplate.setColor('RANDOM');

			const data = this.paginateArray(this.items, i + 1, this.itemsPerPage);
			this.addPage({
				embeds: [this.embedTemplate.addField(this.fieldTitle, data.join('\n'), false).addFields(fieldsClone)]
			});
		});
```

Also move that instanceof check because I removed it here